### PR TITLE
ISS-704: Add missing operator tests.

### PR DESCRIPTION
**Pull request by nhaydel**
Steps to test:

1. Unit tests pass
2. Tests exist for the following operators:
```
is_complete_date
exists
additional_columns_empty
variable_metadata_equal_to
is_contained_by
is_contained_by_case_insensitive
prefix_is_contained_by
suffix_is_contained_by
is_unique_set
is_unique_relationship
has_next_corresponding_record
is_ordered_set
is_ordered_by
target_is_sorted_by
value_has_multiple_references
conformant_value_data_type
conformant_value_length
references_correct_codelist
uses_valid_codelist_terms
```

### CODE REVIEW

**Review by RamilCDISC — COMMENTED** [review]
While the code correctly implements the AC, but I have noticed that tests for following operatiors are missing still

1. additional_columns_empty
2. variable_metadata_equal_to
3. prefix_is_contained_by
4. suffix_is_contained_by

File: tests/unit/test_check_operators/test_metadata_checks.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    vlm = [{"filter": filter_func, "type_check": type_check}]
+    df = dataset_type.from_dict(data)
+    result = DataframeType(
+        {"value": df, "value_level_metadata": vlm}
+    ).conformant_value_data_type({})
+    assert result.equals(df.convert_to_series([True, False, False]))
+
+
+@pytest.mark.parametrize("dataset_type", [PandasDataset, DaskDataset])
+def test_variable_metadata_equal_to(dataset_type):
```
Comment (nhaydel) [inline tests/unit/test_check_operators/test_metadata_checks.py:53]: @RamilCDISC I will add the others, but the test for variable_metadata_equal_to is here

Comment (RamilCDISC) [reply tests/unit/test_check_operators/test_metadata_checks.py:53]: @nhaydel Thank you Nic.

**Review by RamilCDISC — APPROVED** [review]
Tests for all the missing operators mentioned in the AC has been added. All tests were reviewed to validate if they properly tests the functionality including checks for positive, negative and edge cases.
